### PR TITLE
GUNDI-4109: `sensor_featured_properties` approach (with unit tests)

### PR DESCRIPTION
**Pull request by vgarcia13**
### Relevant Link:

https://allenai.atlassian.net/browse/GUNDI-4109

---

This pull request introduces support for configuring and displaying "featured" sensor properties in observation data, along with comprehensive unit tests for the client and handler logic. 

The changes allow users to specify which sensor readings should be highlighted, and ensure that these are properly parsed, processed, and included in the output. Additionally, new and expanded tests improve reliability and coverage for the client and handler modules.

**Support for featured sensor properties:**

* Added a new optional `sensor_featured_properties` field to `PullObservationsConfig` and `PullSensorObservationsPerStationConfig` to allow configuration of which sensor data should be displayed as "featured" (`app/actions/configurations.py`).
* Implemented parsing logic for the featured properties and integrated it into the observation pulling and transformation process, ensuring only the configured readings are highlighted in the output (`app/actions/handlers.py`) [[1]](diffhunk://#diff-f575ea29ea1c9649ccb87f9c38a0fd9a78b9f210c9e5e9d800e328ee30d08113L29-R40) [[2]](diffhunk://#diff-f575ea29ea1c9649ccb87f9c38a0fd9a78b9f210c9e5e9d800e328ee30d08113R61-R70) [[3]](diffhunk://#diff-f575ea29ea1c9649ccb87f9c38a0fd9a78b9f210c9e5e9d800e328ee30d08113L60-R81) [[4]](diffhunk://#diff-f575ea29ea1c9649ccb87f9c38a0fd9a78b9f210c9e5e9d800e328ee30d08113R109-R111) [[5]](diffhunk://#diff-f575ea29ea1c9649ccb87f9c38a0fd9a78b9f210c9e5e9d800e328ee30d08113R147) [[6]](diffhunk://#diff-f575ea29ea1c9649ccb87f9c38a0fd9a78b9f210c9e5e9d800e328ee30d08113R156) [[7]](diffhunk://#diff-f575ea29ea1c9649ccb87f9c38a0fd9a78b9f210c9e5e9d800e328ee30d08113L174-R200).

**Testing improvements:**

* Added a new test suite for the client module, covering token retrieval, project fetching, and sensor reading retrieval with various success and error scenarios (`app/actions/tests/test_client.py`).
* Added a new test suite for the handlers module, testing observation pulling and sensor observation extraction, including edge cases and error handling (`app/actions/tests/test_handlers.py`).
* Configured pytest to look for tests in the correct directory (`pytest.ini`).

### How it looks?

[Config](https://stage.gundiservice.org/connections/f531f031-fa74-48f2-a85a-df3d75553a63/provider):

image

---

[ER](https://gundi-dev.staging.pamdas.org/layers):

image

### CODE REVIEW

File: pytest.ini [hunk 1]
```diff
@@ -0,0 +1,2 @@
+[pytest]
+testpaths = app/actions/test
```
[the remark this answers is not in the document]
Comment (chrisdoehring) [reply pytest.ini:2]: @vgarcia13 can you double check this one from copilot?

**Review by chrisdoehring — APPROVED** [review]
@vgarcia13 there is one copilot comment about the test path.

So long as you're happy with that, go ahead and merge this one.
